packages/hs-react-ui/src/components/Button/Button.tsx

```diff
@@ -1,66 +1,104 @@
 import React from 'react';
+import UnstyledIcon from '@mdi/react';
+import { mdiLoading } from '@mdi/js';
 import styled, { StyledComponentBase } from 'styled-components';
-import colors from '../../enums/colors';
-import fonts from '../../enums/fonts';
+import ButtonContainer, { ButtonTypes, ButtonContainerProps } from './ButtonContainers';
+import Progress from '../Progress/Progress';
 
-const Text = styled.span``;
+export type ButtonProps = {
+  StyledContainer?: string & StyledComponentBase<any, {}, ButtonContainerProps>;
+  iconPrefix?: string | JSX.Element;
+  iconSuffix?: string | JSX.Element;
+  isLoading?: boolean;
+  isProcessing?: boolean;
+  children?: string | Node;
+  elevation?: number;
+  type?: string;
+  color?: string;
+  onClick: (...args: any[]) => void;
+  LoadingBar?: string & StyledComponentBase<any, {}>;
+};
 
-export enum ButtonTypes {
-  primary = 'primary',
-  secondary = 'secondary',
-  default = 'default',
-  destructive = 'destructive',
-}
+const Icon = styled(UnstyledIcon)``;
 
-export const ButtonContainer = styled.button`
+const Text = styled.span`
   display: inline-block;
+  margin-top: -8px;
+  margin-bottom: -8px;
+`;
 
-  ${fonts.body}
-  font-size: 1em;
-  line-height: 0;
-
-  padding: 1.25em 1em;
-  border-radius: 0.25em;
-
-  outline: 0 none;
-  border: 0 none;
-
-  cursor: pointer;
+export const StyledProgress = styled(Progress)`
+  width: 5rem;
+  height: 10px;
+  margin-top: -5px;
+  margin-bottom: -5px;
+`;
 
-  background-color: ${colors.grayXlight};
-  color: ${colors.grayDark};
+const LeftIconContainer = styled(Text)`
+  margin-right: 0.25rem;
 `;
 
-export type ButtonProps = {
-  StyledContainer?: string & StyledComponentBase<any, {}>;
-  icon?: any;
-  isLoading?: any;
-  children?: string | Node;
-  onClick: (...args: any[]) => void;
-};
+const RightIconContainer = styled(Text)`
+  margin-left: 0.25rem;
+`;
 
 const Button = ({
   StyledContainer = ButtonContainer,
-  icon,
+  iconPrefix,
+  iconSuffix,
   isLoading,
+  isProcessing,
   children,
+  elevation = 0,
+  type = ButtonTypes.fill,
+  color,
   onClick,
+  LoadingBar = StyledProgress,
 }: ButtonProps) => {
+  const Container = StyledContainer || ButtonContainer;
   return isLoading ? (
-    <StyledContainer>
-      <Text>
-        {isLoading.children} {isLoading.icon}
-      </Text>
-    </StyledContainer>
+    <Container data-test-id="hsui-button" elevation={elevation} color={color} type={type}>
```

Comment:
I think it should still have an onClick - if they don't want to pass it until loading is false that's on them - but we shouldn't force that business logic on them